zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java

```diff
@@ -680,6 +684,7 @@ public String getStrictTransport() {
     ZEPPELIN_ALLOWED_ORIGINS("zeppelin.server.allowed.origins", "*"),
     ZEPPELIN_ANONYMOUS_ALLOWED("zeppelin.anonymous.allowed", true),
     ZEPPELIN_CREDENTIALS_PERSIST("zeppelin.credentials.persist", true),
+    ZEPPELIN_CREDENTIALS_ENCRYPT_KEY("zeppelin.credentials.encryptKey", null),
```

Comment:
does this mean we need to save the entire key in config?

Reply:
you need the private key somewhere accessible by the process, yes. Since these configs also load from the OS env, you can (should?) keep the key there, instead of the .sh file. That's how 12-factor environments deal with these kinds of keys, at least.

Addressing @Leemoonsoo question above - if someone gets hold of your system config, there's basically nothing you can do to prevent them from accessing private keys or any other environment config you might have on your server. You can get better security by using a third-party PCI-compliant "service" to store keys, for instance, but that kind of solution is highly context-dependent and hard to implement generically

Reply:
Thanks for the explanation.